### PR TITLE
[fix] - unhandled producer promise

### PR DESCRIPTION
**Pull request by jonaphael**
Handling Messenger Producer "produce" promise.

### CODE REVIEW

**Comment by mprevide** [comment]
I know other problem with this lib:

in lib/kafka/Producer.js:29

change 
this.producer.on("delivery-report", this._resolveOnDeliveryReport);
to
const resolveOnDeliveryReportBind = this._resolveOnDeliveryReport.bind(this);
this.producer.on("delivery-report", this._resolveOnDeliveryReport);

Check here:

https://github.com/dojot/dojot-microservice-sdk-js/blob/fc225d29309c478dd270858373972336dc9f73bd/lib/kafka/Producer.js#L258
https://github.com/dojot/dojot-microservice-sdk-js/blob/fc225d29309c478dd270858373972336dc9f73bd/lib/kafka/Producer.js#L99

In fact, there are other problems, but... I think we can focus on just that.

File: lib/messenger.js [hunk 1]
```diff
@@ -475,7 +475,11 @@ class Messenger {
       return;
     }
 
-    this.producer.produce(this.producerTopics[subject][tenant], message, key, partition);
+    this.producer.produce(this.producerTopics[subject][tenant], message, key, partition).then( _response => {
+      logger.info(`Message produced successfully`, TAG);
```
Comment (mprevide) [inline lib/messenger.js:479]: change this to debug

Comment (mprevide) [reply lib/messenger.js:479]: or you can use just a try and catch

File: lib/messenger.js [hunk 2]
```diff
@@ -475,7 +475,11 @@ class Messenger {
       return;
     }
 
-    this.producer.produce(this.producerTopics[subject][tenant], message, key, partition);
+    this.producer.produce(this.producerTopics[subject][tenant], message, key, partition).then( _response => {
```
Comment (mprevide) [inline lib/messenger.js:478]: It is ``logger.info (` Message successfully produced`, TAG); `` code actually executed? it could be a debug.

Comment (mprevide) [reply lib/messenger.js:478]: You can change to try/catch like in here: https://github.com/Blizzard/node-rdkafka#standard-api

Comment (jonaphael) [reply lib/messenger.js:478]: I think that resolving and rejecting the promise is cleaner, and that was the idea, but I can change it if necessary.

Comment (mprevide) [reply lib/messenger.js:478]: Soo.. my point is `this.producer.produce`  a promise?

Comment (mprevide) [reply lib/messenger.js:478]: My point is.. Is `produce` a promise?

Comment (mprevide) [reply lib/messenger.js:478]: Ok.. 
I made a confusion between the produce from node-rdkafka and from the lib... 
Sorry
but It could be change for .debug.

**Review by mprevide — APPROVED** [review]

**Pull request merged**